DataService.js

```diff
@@ -57,28 +56,23 @@ angular.module('PathOfDamage')
               total: 0
             },
             less: {
-              placeholder: "Less Taken",
-              values: []
+              placeholder: "Less Taken"
             }
           }
         }
       };
     },
-    generateDataList: function (table) {
-      table = table.slice(0, -1);
+    serializeDataTable: function (table) {
       var data = [];
-      for (var i = 0; i < table.length; i++) {
+      for (var i = 0; i < table.slice(0, -1).length; i++) {
```

Comment:
This should probably just be length - 1 instead of slicing the end off just to calculate length.